frontend/src/Utils/TreeUtils.ts

```diff
@@ -79,7 +79,42 @@ export class TreeUtils {
     });
     return roots;
   }
-  
+
+  private wmsLayersToTreeNodeList(list:any[]):TreeNodeType[] {
+    const roots:TreeNodeType[] = [];
+
+    // initialize children on the list element
+    list = list.map((element: any) => ({ ...element, children: [] }));
+
+    list.map((element) => {
+      const node: TreeNodeType = {
+        key: element.id,
+        title: element.attributes.title,
+        parent: element.relationships.parent.data?.id,
+        children: element.children || [],
+        isLeaf: element.children && element.children.length === 0,
+        properties: {
+          title: element.attributes.title, // yes, title is repeated
+          scaleMin: element.attributes.scale_min,
+          scaleMax: element.attributes.scale_max,
+        },
+        expanded: true
+      };
+
+      if (node.parent) {
+        const parentNode: MPTTJsonApiTreeNodeType | undefined = list.find((el:any) => el.id === node.parent);
+        if (parentNode) {
+          list[list.indexOf(parentNode)].children?.push(node);
+        }
+      } else {
+        roots.push(node);
+      }
+      return element;
+    });
+
+    return roots;
+  }
```

Comment:
This method is exactly the same as the `mapContextLayersToTreeNodeList`. The only thing different are the properties that are passed. I suggest we change this method later to accept this properties as function parameters instead of repeating the code. Let's decide about it next week